crates/tests/src/rpc/read.rs

```diff
@@ -0,0 +1,1038 @@
+#[cfg(test)]
+mod test_rpc_read_calls {
+    use once_cell::sync::Lazy;
+    use std::any::Any;
+    use std::sync::Arc;
+
+    use crate::{MadaraCmd, MadaraCmdBuilder};
+    use flate2::read::GzDecoder;
+    use starknet_core::types::{
+        BlockHashAndNumber, BlockId, BlockStatus, BlockWithReceipts, BlockWithTxHashes, BlockWithTxs,
+        ComputationResources, ContractClass, ContractStorageDiffItem, DataAvailabilityResources, DataResources,
+        DeclareTransaction, DeclareTransactionReceipt, DeclareTransactionV0, EmittedEvent, EventFilter, EventsPage,
+        ExecutionResources, ExecutionResult, FeePayment, FunctionCall, L1DataAvailabilityMode, L1HandlerTransaction,
+        MaybePendingBlockWithReceipts, MaybePendingBlockWithTxHashes, MaybePendingBlockWithTxs,
+        MaybePendingStateUpdate, PriceUnit, ReceiptBlock, ResourcePrice, StateDiff, StateUpdate, StorageEntry,
+        Transaction, TransactionExecutionStatus, TransactionFinalityStatus, TransactionReceipt,
+        TransactionReceiptWithBlockInfo, TransactionStatus, TransactionWithReceipt,
+    };
+    use starknet_core::types::{
+        BroadcastedDeployAccountTransaction, BroadcastedDeployAccountTransactionV1, BroadcastedTransaction, EthAddress,
+        FeeEstimate, Felt, MsgFromL1, SimulationFlagForEstimateFee,
+    };
+    use starknet_providers::Provider;
+    use std::fmt::Write;
+    use std::fs::File;
+    use std::io::BufReader;
+    use std::io::Read;
+    use tokio::sync::{Mutex, OnceCell};
+
+    static MADARA: Lazy<OnceCell<Arc<Mutex<MadaraCmd>>>> = Lazy::new(OnceCell::new);
+
+    async fn setup_madara() -> Arc<Mutex<MadaraCmd>> {
```

Comment:
this could be a fixture. do we need Arc or Mutex? does a #[once] fixture autmatically handle it?

Reply:
resolved